include/up-client-vsomeip-cpp/routing/SomeipRouter.hpp

```diff
@@ -0,0 +1,253 @@
+     * @param[in] umsg
+     *
+     * @return true if outbound message is routed successfully and false otherwise
+    */
+    bool routeOutboundMsg(const uprotocol::utransport::UMessage &umsg);
+
+    /**
+     * @brief This function is used to route inbound subscription/unsubscription
+     * request to MessageRouter for further handling.
+     *
+     * @param[in] strTopic Topic to subscribe
+     * @param[in] isSubscribe true if subscription request and false if unsubscription request
+     *
+     * @return true if subscription request is successful, false otherwise
+    */
+    bool routeInboundSubscription(std::string const &strTopic, bool const &isSubscribe) const override;
+
+    /**
+     * @brief Handles a state change event.
+     *
+     * @note This method is called when a state change event occurs.
+     *
+     * @param[in] currentState The new state of the SomeipRouter.
+     */
+    void onState(state_type_e const currentState);
+
+    /**
+     * @brief @see @ref SomeipRouterInterface::isStateRegistered()
+     *
+     * @return True if the state is registered, false otherwise.
+     */
+    bool isStateRegistered() override;
+
+private:
+
+    /**
+     * @brief Stop the service.
+     *
+     * @note This function stops the execution of the service.
+     * It performs any necessary cleanup or shutdown procedures.
+     * After calling this function, the service is no longer running.
+     */
+    void stopService();
+
+    /**
+     * @brief enableAllLocalServices fetch all local services and calls offerServicesAndEvents
+    */
+    void enableAllLocalServices();
+
+    /**
+     * @brief offerServicesAndEvents offers services and events for the given uri
+    */
+    void offerServicesAndEvents(std::shared_ptr<uprotocol::v1::UUri>);
+
+    /**
+     * @brief getUriList fetches the uri list for the given service type from json
+     * @param serviceType - can be either "services" or "remoteServices"
+     * @return list of UUri as a shared pointer to a vector
+    */
+    std::shared_ptr<std::vector<std::shared_ptr<uprotocol::v1::UUri>>> getUriList(const std::string &serviceType);
+
+    /**
+     * @brief Retrieves the SomeipHandler associated with the given ID.
+     *
+     * @param[in] UEId The unique identifier of the handler.
+     * @return A shared pointer to the SomeipHandler object, or nullptr if no handler is found.
+     */
+    std::shared_ptr<SomeipHandler> getHandler(uint16_t UEId);
+
+    /**
+     * @brief Creates a new SomeipHandler object of the specified type and associates it with the given ID.
+     *
+     * @note This method creates a new SomeipHandler object based on
+     * the specified handler type and associates it with the provided ID
+     *
+     * @param[in] type The type of the handler to create.
+     * @param[in] UEId The unique identifier of the handler.
+     * @return A shared pointer to the newly created SomeipHandler object.
+     */
+    std::shared_ptr<SomeipHandler> newHandler(
```

Comment:
Not sure why we need to instantiate handlers for client or server mode here, maybe you can explain offline.